src/analysisd/decoders/decode-xml.c

```diff
@@ -783,51 +799,102 @@ int ReadDecodeXML(const char *file)
     return retval;
 }
 
-int SetDecodeXML()
+int SetDecodeXML(char **msg)
 {
+    char* tmp_msg = NULL;
     /* Add rootcheck decoder to list */
-    addDecoder2list(ROOTCHECK_MOD);
-    addDecoder2list(SYSCHECK_MOD);
-    addDecoder2list(SYSCHECK_NEW);
-    addDecoder2list(SYSCHECK_DEL);
-    addDecoder2list(HOSTINFO_NEW);
-    addDecoder2list(HOSTINFO_MOD);
-    addDecoder2list(SYSCOLLECTOR_MOD);
-    addDecoder2list(CISCAT_MOD);
-    addDecoder2list(WINEVT_MOD);
-    addDecoder2list(SCA_MOD);
+    addDecoder2list(ROOTCHECK_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(SYSCHECK_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(SYSCHECK_NEW, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(SYSCHECK_DEL, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(HOSTINFO_NEW, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(HOSTINFO_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(SYSCOLLECTOR_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(CISCAT_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(WINEVT_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
+    addDecoder2list(SCA_MOD, &tmp_msg);
+    if(tmp_msg){
+        wm_strcat(msg, tmp_msg, 0);
+        os_free(tmp_msg);
+    }
 
     /* Set ids - for our two lists */
     if (!os_setdecoderids(NULL)) {
-        merror(DECODER_ERROR);
+        smerror(&tmp_msg, DECODER_ERROR);
+        if(tmp_msg){
+            wm_strcat(msg, tmp_msg, 0);
+            os_free(tmp_msg);
+        }
         return (0);
     }
     if (!os_setdecoderids(ARGV0)) {
-        merror(DECODER_ERROR);
+        smerror(&tmp_msg, DECODER_ERROR);
+        if(tmp_msg){
+            wm_strcat(msg, tmp_msg, 0);
+            os_free(tmp_msg);
+        }
         return (0);
     }
 
     return (1);
 }
 
-/* Allocate memory at "*at" and copy *str to it
- * If *at already exist, realloc the memory and cat str on it
- * Returns the new string
+/**
```

Comment:
Doxygen documentation should stay in the declaration of the header file (line 21)